ops/framework.py

```diff
@@ -518,7 +518,10 @@ def __getattr__(self, name: str) -> BoundEvent:
 
 
 class LifecycleEvent(EventBase):
-    """Events tied to the lifecycle of the Framework object."""
+    """Events tied to the lifecycle of the Framework object.
+
+    Note: it is totally different from Juju [application] lifecycle.
+    """
```

Comment:
Small punctuation and rephrase suggestion:

> Note: Here, "lifecycle" has nothing to with Juju [application] lifecycle.

Reply:
+1 on the rephrasing, but wouldn't it be better as "the Juju"?

Also, maybe a link to the Juju docs to help understand the difference?

Reply:
I omitted Juju link thinking that having a link would be more confusing, LOL!

Reply:
rephrased as suggested, not adding an external link.